src/partitioning/metis_partitioner.C

```diff
@@ -403,8 +403,16 @@ void MetisPartitioner::_do_partition (MeshBase & mesh,
                     // would be nice
                     Elem * neighbor = const_cast<Elem *>(*n_it);
 
-                    csr_graph(elem_global_index, connection++) =
-                      global_index_map[neighbor->id()];
+                    // Not all interior neighbors are necessarily in
+                    // the global_index_map.  This will be the case
+                    // when partitioning a BoundaryMesh, whose
+                    // elements all have interior_parents() that
+                    // belong to some other Mesh.
+                    vectormap<dof_id_type, dof_id_type>::iterator global_index_map_it =
+                      global_index_map.find(neighbor->id());
+
```

Comment:
In my case, the elements of the `BoundaryMesh` were numbered in a non-overlapping way from their `interior_parents()`, but I suppose that won't be the case in general, especially if it later gets renumbered.  I'll make a more thorough fix...